backend/src/controllers/MyRestaurantController.ts

```diff
@@ -40,7 +37,45 @@ const getMyRestaurant = async (req: Request, res: Response) => {
   }
 };
 
+const updateMyRestaurant = async (req: Request, res: Response) => {
+  try {
+    const restaurant = await Restaurant.findOne({ user: req.userId });
+    if (!restaurant) {
+      return res.status(404).json({ message: 'Restaurant not found' });
+    }
+
+    restaurant.restaurantName = req.body.restaurantName;
+    restaurant.city = req.body.city;
+    restaurant.country = req.body.country;
+    restaurant.deliveryPrice = req.body.deliveryPrice;
+    restaurant.estimatedDeliveryTime = req.body.estimatedDeliveryTime;
+    restaurant.cuisines = req.body.cuisines;
+    restaurant.menuItems = req.body.menuItems;
+    restaurant.lastUpdated = new Date();
+
+    if (req.file) {
+      const imageUrl = await uploadImage(req.file as Express.Multer.File);
+      restaurant.imageFile = imageUrl;
+    }
+
+    await restaurant.save();
+    res.status(200).json(restaurant);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({ message: 'Error updating restaurant' });
+  }
+};
+
+const uploadImage = async (file: Express.Multer.File) => {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/34#issuecomment-2394960890>#30a71e</a><span> - 10/05/2024, 07:22 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The uploadImage function lacks proper file validation and error handling.
</div>
<div id="fix">
<b>Fix:</b> Implement file type and size validation, and add proper error handling in the uploadImage function.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -69,8 +69,20 @@ const uploadImage = async (file: Express.Multer.File) => {
    const image = file;
 +  // Validate file type
 +  const allowedTypes = ['image/jpeg', 'image/png', 'image/gif'];
 +  if (!allowedTypes.includes(image.mimetype)) {
 +    throw new Error('Invalid file type. Only JPEG, PNG, and GIF are allowed.');
 +  }
 +
 +  // Validate file size (e.g., max 5MB)
 +  const maxSize = 5 * 1024 * 1024; // 5MB
 +  if (image.size > maxSize) {
 +    throw new Error('File size exceeds the limit of 5MB.');
 +  }
 +
    const base64Image = image.buffer.toString('base64');
    const dataURI = `data:${image.mimetype};base64,${base64Image}`;
 -  const uploadResponse = await cloudinary.uploader.upload(dataURI);
 -  return uploadResponse.secure_url;
 +  try {
 +    const uploadResponse = await cloudinary.uploader.upload(dataURI);
 +    return uploadResponse.secure_url;
 +  } catch (error) {
 +    console.error('Error uploading image to Cloudinary:', error);
 +    throw new Error('Failed to upload image. Please try again.');
 +  }
  };
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged